### Задача №1, 2

Контрольная работа состоит из пяти вопросов. На каждый вопрос приведено четыре варианта ответа, один из которых правильный. Случайная величина X задаёт число правильных ответов при простом угадывании. Найдите математическое ожидание данной случайной величины. Найдите дисперсию случайной величины X.

In [17]:
import numpy as np

In [27]:
n = 5
p = 0.25

In [28]:
mat = n * p
dis = n * p * (1 - p)

In [29]:
print(mat, dis)

1.25 0.9375


Ответ: М(Х) = 1.25, D(X) = 0.9375

### Задача №3

Пользователь получает в среднем 10 писем со спамом на свой почтовый ящик за сутки. Найти число N, такое, что с вероятностью 0.95 пользователь получит не более N писем со спамом за текущий день.

In [18]:
def poisson_proba(k: int, lambda_: float) -> float:
    return (lambda_ ** k) * np.exp(-lambda_) / np.math.factorial(k)

In [19]:
lambda_ = 10
p = 0
n = 0

In [20]:
while p < 0.95:
    n += 1
    p += poisson_proba(n, lambda_)
    print(n, p)


1 0.00045399929762484856
2 0.002723995785749091
3 0.010290650746163234
4 0.029207288147198588
5 0.0670405629492693
6 0.1300960209527205
7 0.2201752466719365
8 0.33277427882095645
9 0.4578843145420898
10 0.5829943502632231
11 0.6967307463733443
12 0.791511076465112
13 0.8644190226895486
14 0.9164961271355748
15 0.951214196766259


Ответ: N = 15

### Задача №4

Производятся выстрелы по мишени. Вероятность попадания в мишень при одном выстреле равна 0.01. Сколько выстрелов нужно сделать чтобы быть уверенным с вероятностью 0.9, что хотя бы 6 раз будет совершено попадание?

In [21]:
def binominal(n: int, k: int, p: float):
    return np.math.factorial(n) / (np.math.factorial(k) * np.math.factorial(n - k)) * (p ** k) * ((1 - p) ** (n - k))

In [22]:
n = 5
p = 0.01
prop = 0

In [23]:
while prop < 0.9:
    n += 1
    prop = 1 - sum(binominal(n, k, p) for k in range(6))
    
print(n, prop)

926 0.9003627121531355


Ответ: необходимо 926 выстрелов.